mono/mini/mini-llvm.c

```diff
@@ -7341,6 +7341,27 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			break;
 		}
 		case OP_XEQUAL: {
+#if defined(TARGET_AMD64) || defined(TARGET_X86)
+			int nelemsi8 = 16; // TODO: calculate vector width in bytes for current CPU
+			LLVMValueRef lhs8 = LLVMBuildBitCast (ctx->builder, lhs, LLVMVectorType (LLVMInt8Type (), nelemsi8), "");
+			LLVMValueRef rhs8 = LLVMBuildBitCast (ctx->builder, rhs, LLVMVectorType (LLVMInt8Type (), nelemsi8), "");
+
+			//%5 = icmp eq <16 x i8> %3, %4
+			LLVMValueRef cmp = LLVMBuildICmp (builder, LLVMIntEQ, lhs8, rhs8, "");
```

Comment:
`pcmpeqb` is not a floating point equality comparison, and will not correctly handle NaNs or negative zero.